src/lexer/Tokenizer.ts

```diff
@@ -73,7 +73,7 @@ export default class Tokenizer {
       [TokenType.QUOTED_IDENTIFIER]: { regex: regex.string(cfg.identTypes) },
       [TokenType.NUMBER]: {
         regex:
-          /(?:0x[0-9a-fA-F]+|0b[01]+|(?:-\s*)?[0-9]+(?:\.[0-9]*)?(?:[eE][-+]?[0-9]+(?:\.[0-9]+)?)?)/uy,
+          /(?:0x[0-9a-fA-F]+|0b[01]+|(?:-\s*)?[0-9]+(?:\.[0-9]*)?(?:[eE][-+]?[0-9]+(?:\.[0-9]+)?)?)(?!\w)/uy,
```

Comment:
Why the negative lookahead instead of just `\b`?

Reply:
\b would match `[1]two3` whereas the negative lookahead prevents it from being matched at all so the identifier pattern can match later in the priority

Reply:
Sorry, I don't understand. What do you mean by `[1]two3`?

The `\b` (word boundary) does not include any extra text inside the match, just like negative lookahead.

Reply:
using \b would mean "match if followed by a non-word character"
whereas the negative lookahead is "match unless followed by a word character"

Reply:
the `1` in `1two3` would get tokenized as a number before reaching the identifier pattern, leaving it as `1 (number) two3 (identifier)`

Reply:
These two statements are logically equivalent.

Although `\b` does not refer to non-word character (that would be `\W`). It refers to word-boundary e.g. being preceded by word character and followed by non-word character, or the other way around.

Reply:
the first statement is A if not B whereas the second is not(A if B)
simply: the \b _does_ match the leading digit before reaching the identifier regex (undesired behaviour) and the negative lookahead does not match so the identifier pattern can match (desired behaviour)

Reply:
Well, there is no word-boundary between `1` and `two3`. Both numbers and letters are considered to be word characters.

There is though one case where the simple `\b` in the end doesn't work: a number ending with a period like `100.` followed by space. In that case there wouldn't be a word-boundary at the end of the number. So, to cover this ends-with-period case, the negative lookahead is probably a better choice here.